### PR TITLE
Created BoxTrackingCode model and updated BoxUser and BoxUserRequest …

### PR DESCRIPTION
**Pull request by allenmichael**
…to use this model.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @allenmichael has signed the CLA. Thanks for the pull request!

**Comment by allenmichael** [comment]
Not sure what's going on here. Completely unrelated unit tests to the code I touched are failing in AppVeyor.

File: Box.V2/Models/BoxUser.cs [hunk 1]
```diff
@@ -88,7 +89,7 @@ public class BoxUser : BoxEntity
         /// An array of key/value pairs set by the user’s admin
         /// </summary>
         [JsonProperty(PropertyName = FieldTrackingCodes)]
-        public string[] TrackingCodes { get; private set; }
```
Comment (carycheng77) [inline Box.V2/Models/BoxUser.cs]: This might be a breaking change, would love to chat and grab more context before we merge, thanks!

**Comment by mattwiller** [comment]
Superseded by #459

**Pull request closed without merging**